moveit_ros/planning/planning_scene_monitor/src/trajectory_monitor.cpp

```diff
@@ -44,7 +45,16 @@ static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_ros.planning_sce
 
 planning_scene_monitor::TrajectoryMonitor::TrajectoryMonitor(const CurrentStateMonitorConstPtr& state_monitor,
                                                              double sampling_frequency)
+  : TrajectoryMonitor(state_monitor, std::make_unique<TrajectoryMonitorMiddlewareHandle>(sampling_frequency),
+                      sampling_frequency)
+{
+}
+
+planning_scene_monitor::TrajectoryMonitor::TrajectoryMonitor(
+    const CurrentStateMonitorConstPtr& state_monitor,
+    std::unique_ptr<TrajectoryMonitor::MiddlewareHandle> middleware_handle, double sampling_frequency)
   : current_state_monitor_(state_monitor)
+  , middleware_handle_(std::move(middleware_handle))
```

Comment:
Consider (though a separate PR also makes sense) converting to uniform initialization

Reply:
```suggestion
  , middleware_handle_{std::move(middleware_handle)}
```

:arrow_up: that is all Griz means btw.

Reply:
Yes, but for all